drivers/led/leds_gpio.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018 Manivannan Sadhasivam
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief LEDs driver for GPIOs
+ *
+ * This driver controls LED devices attached to GPIO pins which
+ * are declared under leds node in board devicetree. The driver
+ * takes care of configuring the pin to output state before using it.
+ *
+ * TODO: Implement Software Blinking
+ */
+
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+#include <led.h>
+#include <misc/util.h>
+#include <zephyr.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_LED_LEVEL
+#include <logging/sys_log.h>
+
+#include "led_context.h"
+
+enum led_gpio_modes {
+	LED_OFF,
+	LED_ON,
+};
+
+struct led_gpio_cfg {
+	char *gpio_port;
+	u8_t gpio_pin;
+};
+
+struct led_gpio_data {
+	struct device *gpio;
+	struct led_data dev_data;
+};
+
+static inline int led_gpio_on(struct device *dev, u32_t led)
```

Comment:
Is 'led' argument actually required ?
We should be able to retrieve it from *dev. 
And I think this is the interest of the API, so user doesn't care of the PIN

Reply:
`led` argument is there only for the purpose of LED API and yes, we can retrieve it from *dev. Anyway, passing the `led` argument gives us the possibility of checking the correct pin in led_on/led_off as below:

```c
        /* Make sure that the LED is valid */
        if (led != cfg->gpio_pin)
                return -EINVAL;
```
Will include the above check.

Reply:
@Mani-Sadhasivam , 
> Anyway, passing the led argument gives us the possibility of checking the correct pin in led_on/led_off

In the other hand not having 'led' as arguments makes this kind of issue impossible.
I still prefer this is removed from the API.

Reply:
I agree the argument should stay and the driver should check it.

Reply:
Can you check following PR to see if it wouldn't provide similar service and spare the argument?
https://github.com/zephyrproject-rtos/zephyr/pull/12056

Reply:
I don't see how #12056 relates.  I also agree that since this is being done as an implementation for an API that is designed to support multiple LEDs the argument must stay.  However, see review summary.